src/components/Trait/WardrobeTraitSelector.tsx

```diff
@@ -43,17 +44,50 @@ const WardrobeTraitSelector: React.FC<Props> = ({ categories }) => {
 		}
 	}
 
+	const hasOnlyOneClothingItem = (metadataTraits: Trait[]) => {
+		const clothingCountSets = metadataTraits.filter(t =>
+			atLeastSomeClothes.some(category => t.categories?.includes(category)),
+		).length
+		const clothingCountItems = metadataTraits.filter(
+			t => t.categories?.includes('Tops') || t.categories?.includes('Bottoms'),
+		).length
+
+		return (
+			(clothingCountSets === 1 && clothingCountItems === 0) ||
+			(clothingCountSets === 0 && clothingCountItems === 2)
+		)
+	}
+
 	const updateTrait = (
 		metadataTraits: Trait[],
 		categoryString: string,
 		value: string,
 	) => {
 		const hadMatch = metadataTraits.find(t => t.name === value)
-		const updatedMetadata = metadataTraits.filter(
-			t => t.categories?.join(' - ') !== categoryString,
-		)
+		const mustHaveExactlyOne = oneAndOnlyOne.includes(categoryString)
+		const islastPieceOfClothing =
+			hadMatch && hasOnlyOneClothingItem(metadataTraits)
+		let updatedMetadata = [...metadataTraits]
+
+		if ((hadMatch && mustHaveExactlyOne) || islastPieceOfClothing) {
+			// FIXME How does UI handle not being able to remove a trait
+			return
+		}
+
+		if (mustHaveExactlyOne) {
+			// selected, remove previous
+			updatedMetadata = metadataTraits.filter(
+				t => t.categories?.join(' - ') !== categoryString,
+			)
```

Comment:
This is a real common pattern. If we touch this again, can we refactor somewhere common?